modules/olm-configuring-aws-sts.adoc

```diff
@@ -0,0 +1,138 @@
+// Module included in the following assemblies:
+//
+// * operators/admin/olm-adding-operators-to-cluster.adoc
+// * storage/persistent_storage/persistent-storage-csi-aws-efs.adoc
+// * storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc
+// * storage/container_storage_interface/osd-persistent-storage-aws-efs-csi.adoc
+// * storage/container_storage_interface/rosa-persistent-storage-aws-efs-csi.adoc
+
+:_content-type: PROCEDURE
+[id="efs-sts_{context}"]
+= Configuring an Operator to use Amazon Web Services Secure Token Service
+
+You can configure Operators on Amazon Web Services (AWS) clusters to use the AWS Secure Token Service (STS). Operators that use STS assign components IAM roles that provide short-term, limited-privilege security credentials.
+
+Perform this procedure after installing the an Operator that can use STS, but before installing any required drivers for the Operator. If you perform this procedure after installing a driver for an Operator, the Operator might not be able to function correctly.
+
+[NOTE]
+====
+By default, `ccoctl` creates objects in the directory in which the commands are run. To create the objects in a different directory, use the `--output-dir` flag. This procedure uses `<path_to_ccoctl_output_dir>` to refer to this directory.
+
+Some `ccoctl` commands make AWS API calls to create or modify AWS resources. You can use the `--dry-run` flag to avoid making API calls. Using this flag creates JSON files on the local file system instead. You can review and modify the JSON files and then apply them with the AWS CLI tool using the `--cli-input-json` parameters.
+====
+
+.Prerequisites
+
+* You have an AWS cluster that is configured to use STS.
+
+* You have the {product-title} release image that was used to install the cluster.
+
+* You have extracted and prepared the `ccoctl` binary during cluster installation. For more information, see "Configuring the Cloud Credential Operator utility".
+
+.Procedure
+
+. Determine the name of the `CredentialsRequest` custom resource (CR) in the `openshift-cloud-credential-operator` namespace for the Operator:
+//How do they do that?
```

Comment:
In the next step, they need to refer to the CR in the `openshift-cloud-credential-operator` namespace, but I do not see how to get that string from the CSI or ALB procedures.

Reply:
ACK - sent you some followup in Slack

Reply:
> The CredentialsRequests for the examples (CSI and ALB) were created by the Operator. I am not sure if it's a standard/required for all 3rd-party operators to create it (@abutcher thoughts?). If the Operator does not provide the CredentialsRequests resources, it should be created by the user manually.

In the STS setup for OpenShift operators, we extract the `CredentialsRequests` from the release image to create the required infrastructure ahead of installation. CCO is then ran in Manual mode, meaning there is no ongoing reconciliation of `CredentialsRequests` so I would expect the `CredentialsRequest` would be created by the user manually.

Reply:
per Slack convo, sounds like best strategy will be to just tell everyone how to make creds regardless of what an Operator may or may not provide (esp as minting becomes less relevant). in that case we just need a very clear, easy to fill-in YAML template in here :+1: